src/components/List/List.js

```diff
@@ -17,9 +17,33 @@ const NAVIGATION_STEP_VALUES = {
     [LIST_NAVIGATION_DIRECTIONS.DOWN]: 1
 };
 
+const SCROLL_INTO_VIEW_SETTINGS = {
+    block: 'nearest'
+};
+
 const List = React.forwardRef((props, ref) => {
-    const { children, isDivided, onNavigate, onSelect, isControlledNavigation, ...rest } = props;
     const [selectedIndex, setSelectedIndex] = useState(null);
+    const [highlightedWithKeyboard, setHighlightedWithKeyboard] = useState(false);
+
+    const highlightedListItem = useRef(null);
+
+    const { children, isDivided, onNavigate, onSelect, isControlledNavigation, ...rest } = props;
+
+    /**
+     * Scroll list if it's necessary to make the highlighted item visible
+     * every time selectedIndex was changed with the keyboard navigation
+     */
+    useEffect(() => {
+        const hasScrollIntoViewFunction =
+            highlightedListItem &&
+            highlightedListItem.current &&
+            highlightedListItem.current.scrollIntoView;
+
+        if (highlightedWithKeyboard && hasScrollIntoViewFunction) {
+            highlightedListItem.current.scrollIntoView(SCROLL_INTO_VIEW_SETTINGS);
+            setHighlightedWithKeyboard(false);
+        }
+    }, [selectedIndex]);
 
     const getNextSelectedIndex = keyCode => {
```

Comment:
Looks good. But I've just noticed that all the event handlers in this component are being recreated on every render, which is not efficient. Could you wrap them into `useCallback` in order to memoize it?